paasta_tools/mesos_tools.py

```diff
@@ -615,3 +615,36 @@ def get_mesos_network_for_net(net):
         'host': 'HOST',
     }
     return docker_mesos_net_mapping.get(net, net)
+
+
+def get_mesos_task_count_by_slave(mesos_state, pool='default'):
+    """Get counts of running tasks per mesos slave. Also include separate count of chronos tasks
+
+    :param mesos_state: mesos state dict
+    :param pool: pool of slaves to return (None means all)
+    :returns: dict of {<slave_id>:{'count': <count>, 'slave': <slave_object>, 'chronos_count':<chronos_count>}}
+    """
+    all_mesos_tasks = get_running_tasks_from_active_frameworks('')  # empty string matches all app ids
+    if pool:
+        slaves = {
+            slave['id']: {'count': 0, 'slave': slave, 'chronos_count': 0} for slave in mesos_state.get('slaves', [])
+            if slave['attributes'].get('pool', 'default') == pool
+        }
+    else:
+        slaves = {
+            slave['id']: {'count': 0, 'slave': slave, 'chronos_count': 0} for slave in mesos_state.get('slaves', [])
+        }
+    for task in all_mesos_tasks:
+        if task.slave['id'] not in slaves:
+            log.error("Task associated with %s but slave not in masters list of slaves" % task.slave['id'])
+            continue
+        else:
+            slaves[task.slave['id']]['count'] += 1
+            log.debug("Task framework: {0}".format(task.framework.name))
+            if task.framework.name == 'chronos':
```

Comment:
Ugh, you're right! As discussed, I'm going to tackle this with a puppet change and a PR against Chronos to remove the version from this property rather than make this any more hacky here.